dist-material/bin/webappService.sh

```diff
@@ -37,8 +37,11 @@ eval exec "\"$_RUNJAVA\" ${JAVA_OPTS} -jar ${JAR_PATH}/skywalking-webapp.jar \
          --logging.file=${LOG_FILE_LOCATION} \
         2>${WEBAPP_LOG_DIR}/webapp-console.log 1> /dev/null &"
 
+UI_PID_FILE="${WEBAPP_HOME}/bin/ui.pid"
+
 if [ $? -eq 0 ]; then
-    sleep 1
+  sleep 1
+  /bin/echo -n $! > "$UI_PID_FILE"
```

Comment:
I suggest moving `$!` to line 39 and give it a variable, in case we add background commands between line 38 and this line, the pid will be wrong

Reply:
I don't know how to give it a variable ,because _$! is the PID of the last program your shell ran in the background_

Reply:
> I don't know how to give it a variable ,because _$! is the PID of the last program your shell ran in the background_

I know. I mean `pid=$!`

Reply:
Okay